### PR TITLE
Build for Jazzy with mavros from source

### PR DESCRIPTION
**Pull request by amarburg**
Per [blue!169](https://github.com/Robotic-Decision-Making-Lab/blue/pull/169), an official image for Jazzy is blocked by the lack of a [Jazzy package for Mavros](https://github.com/mavlink/mavros/issues/1958)

## Changes Made

* Install Mavros and Mavlink from source ([Rolling release](https://github.com/mavlink/mavlink-gbp-release/tree/release/rolling/mavlink)) directly into the blue workspace. 
* Updates to Gazebo "Harmonic"

* This also required bashing through issues related to Ubuntu 24.04: 
  * Ubuntu 24.04 now has a "externally managed installation" of Python, which discourages "pip install"ing packages outside of virtualenv.   Create virtual env `~/.venv/blue` and activate by default both within the build and when running in the docker.   This leads to rosdep installing some pip dependencies into the venv.

## Testing

Please provide a clear and concise description of the testing performed.

### CODE REVIEW

**Comment by evan-palmer** [comment]
There's some additional information regarding the move away from global Python package installations in [PEP 668](https://peps.python.org/pep-0668/). From what I understand, Ubuntu will be keeping the marker file present in the base image, and because the ROS images [use](https://github.com/osrf/docker_images/blob/master/ros2/source/devel/Dockerfile) Ubuntu as their base image, this problem will persist. I haven't tried this yet, but you could try removing the marker file from the base image:

```sh
rm -rf /usr/lib/python3.12/EXTERNALLY_MANAGED
```

You might need to updated the system path afterwards as well. Using a virtual environment seems like the best approach moving forward though.

File: .docker/Dockerfile [hunk 1]
```diff
@@ -40,7 +42,7 @@ FROM ci as robot
 
 # Configure a new non-root user
 ARG USERNAME=blue
-ARG USER_UID=1000
+ARG USER_UID=1001   # ros image now includes a user "ubuntu" at UID 1000
```
Comment (evan-palmer) [inline .docker/Dockerfile:45]: You may run into permissions issues here when trying to use the devcontainer. The blue user likely won't belong to the same group as your personal user on your local system (set to 1000 by default). My solution has been to go along with the ubuntu user standardization.

Comment (amarburg) [reply .docker/Dockerfile:45]: Hm, I hadn't considered the permission ramifications.   So you are using the user `ubuntu`?

```
# Configure a new non-root user
#
# ros image now includes a user "ubuntu" at UID 1000
ARG USERNAME=ubuntu
ARG USER_UID=1000  
ARG USER_GID=$USER_UID

RUN echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME \
    && chmod 0440 /etc/sudoers.d/$USERNAME \
    && usermod -a -G dialout $USERNAME \
    && echo "source /usr/share/bash-completion/completions/git" >> /home/$USERNAME/.bashrc
```

Comment (evan-palmer) [reply .docker/Dockerfile:45]: Yeah. This was motivated by discussions in [cloud-images!2005129](https://bugs.launchpad.net/cloud-images/+bug/2005129), which make it seem like the Ubuntu devs are attempting to standardize the `ubuntu` user. You could still delete the `ubuntu` user to regain access to UID 1000, but I opted against that.

**Comment by amarburg** [comment]
@evan-palmer  (finally), I believe this branch is complete.  Multi-platform builds are time consuming.

Starting on testing.

**Comment by amarburg** [comment]
This PR has been split up:

* [blue PR 223](https://github.com/Robotic-Decision-Making-Lab/blue/pull/223) contains the migration from Gazebo Garden to Harmonic (for all releasese)
* [blue PR 220](https://github.com/Robotic-Decision-Making-Lab/blue/pull/220) handles the migration of **rolling** to ubuntu "Noble", which requires handling the "ubuntu" user and python venv.    Once merged it should be propagated to jazzy

Once those two are merged, the jazzy branch then needs to be updated to build Mavros from source (there **are** binary packages for rolling)

**Pull request closed without merging**